### DISCUSSION TITLE
Does haraka support PGP/GPG or S/MIME?

### DISCUSSION THREAD

**Discussion by php-wizard** [haraka/Haraka#3373] [Q&A] [answered]
- [x] I already searched past issues
- [x] I already checked the [wiki/TODO](https://github.com/haraka/Haraka/wiki/TODO) page

### Is your feature request related to a problem? Please describe.

We need a email server with encryption support . Either GPG/PGP or S/MIME (self-signed) , or any alternatives.
Does Haraka or some of its plugins support it  ?

**Comment by msimerson** [comment, marked as answer]
Yes. Both GPG and S/MIME work with Haraka, no plugins needed.

**Comment by monperrus** [comment]
Hi, I would like to set up a paranoid server that would reject all non-encrypted emails. That is, checking at incoming time that all emails are properly encrypted with GPG. How to do this? Thanks!

**Reply by msimerson** [reply]
1. Yuck. 
2. As someone who has used GPG and S/MIME for many years, your email box is going to be fairly lonely and it'll be difficult to keep it working.
3. Email encryption is already handled much better with TLS which has far better interoperability. 

Buy hey, go write yourself a Haraka plugin that inspects the message bodies and rejects any that aren't of the correct MIME type(s).

**Comment by monperrus** [comment]
Plus we have a tutorial for this https://haraka.github.io/tutorials/tutorial which is great! Thanks @msimerson
